### ISSUE TITLE
Rename builtin plugins to modules

### ISSUE THREAD

**Issue by zardoy** [PrismarineJS/flying-squid#631]
I think it can be confusing for new contributors to have both plugins and lib/plugins folder. Maybe rename the first folder to `modules` or `builtinPlugins`?
I also have a plan to migrate the plugin system to TypeScript for massive intellisense. For example:

```ts
//@filename: player.ts
export default definePlugin({
	server(...) {...},
   player(player, serv, { version }) {/* ... */},
})

declare interface API {
	player: ...
}
```

**Comment by rom1504** [comment]
I don't think any of these changes would help much and it would cause confusion by making the change in itself

**Comment by zardoy** [comment]
> I don't think any of these changes would help much and it would cause confusion by making the change in itself

can you rephrase. not clear

**Comment by zardoy** [comment]
this structure seems less confusing and has a lot of benefits https://github.com/zardoy/space-squid/tree/everything/src/lib such as improved stability & intellisense (i mean its just structured, isnt' it?)

**Comment by rom1504** [comment]
Btw I think it's too bad you keep doing these hard forks (https://github.com/PrismarineJS/flying-squid/compare/master...zardoy:space-squid:everything). The consequences of that is that someday your forks will go unmaintained and it'll be very hard / impossible to merge back into the main branch to build on top of any improvements 

Did you consider proposing incremental improvements instead?

**Comment by zardoy** [comment]
> Btw I think it's too bad you keep doing these hard forks

My bad, sorry! Yes, my soft might be just terrible, I just do what I like to do and there is no point in working on the code for fun if you are not getting any fun.

> The consequence of that is that someday your forks will go unmaintained and it'll be very hard / impossible to merge back into the main branch to build on top of any improvements

Would be really great if you could give me more specific instructions on what would be more acceptable to you and what I can do here.

> Did you consider proposing incremental improvements instead?

Incremental changes are easier to do after doing bigger ones, also I tried here to have clean commits history (yes, sorry for not opening prs, was always planning to do that, but wasn't sure). If you want I can group / clean commit history just like you want. Would be much easier if you tell me how I can pick the changes and in what changes you are interested. And thanks for feedback!
